### PR TITLE
Fix wrong number of maxAttrNum in TupleSplitState

### PR DESCRIPTION
**Pull request by charliettxx**
we cannot get corrent number of projecting targets for executing TupleSplit Node if wrong maxAttrNum in TupleSplitState, that cause wrong results in multi-dqa sql. like sql
```
select count(distinct a), count(distinct b) from dqa_f4 group by c;

For each splited tuple, we also need to project column `c` as group column besieds
distinct column `a` and `b`. As a result, toal maxAttrNum of TupleSplit is three instead of two, 
which also decided totals projection columns of Node Tuplesplit.
```
To figure maxAttrNum correctly, we need to calculate again after we initiated all elements in TupleSplitState.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by CharlieTTXX** [comment]
related issue https://github.com/greenplum-db/gpdb/issues/14688

File: src/test/regress/sql/gp_dqa.sql [hunk 1]
```diff
@@ -541,3 +541,11 @@ explain (verbose on, costs off) select sum(Distinct b), count(c) filter(where c
 select sum(Distinct b), count(c) filter(where c > 1), sum(a) from dqa_f3;
 
 drop table dqa_f3;
+
+-- Test some corner case of dqa ex.NULL
+create table dqa_f4(a int, b int, c int);
+insert into dqa_f4 values(null, null, null);
```
Comment (adam8157) [inline src/test/regress/sql/gp_dqa.sql:547]: Better to insert several not NULL lines too, to verify the result?

**Review by adam8157 — COMMENTED** [review]
The codes look legit, try writing a better RCA in the commit message.

File: src/test/regress/sql/gp_dqa.sql (hunk 1 above)
Comment (CharlieTTXX) [reply src/test/regress/sql/gp_dqa.sql:547]: Good catch, I've done it.

File: src/backend/executor/nodeTupleSplit.c [hunk 2]
```diff
@@ -143,6 +141,18 @@ ExecInitTupleSplit(TupleSplit *node, EState *estate, int eflags)
 		bms_free(orig_bms);
 	}
 
+	/*
+	 * Update maxAttrNum which is used to calculate projection number
+	 * of ExecTupleSplit
+	 */
+	int x = -1;
```
Comment (QingMa-gp) [inline src/backend/executor/nodeTupleSplit.c:148]: `x` can be initialized to `maxAttrNum` here.

Comment (charliettxx) [reply src/backend/executor/nodeTupleSplit.c]: done it.

**Review by QingMa-gp — APPROVED** [review]

**Review by adam8157 — APPROVED** [review]
